src/UriBuilder.Fluent/TerribleDevUriExtensions.cs

```diff
@@ -56,6 +58,56 @@ public static UriBuilder WithParameter(this UriBuilder bld, string key, IEnumera
             return bld;
         }
 
+        public static UriBuilder WithParametersFromObject(this UriBuilder bld, object parameters)
```

Comment:
this method does not handle objects that contain properties that have child properties

`Challenge to you if you choose to accept (and you don't have to, just lmk if you do or not):`

devise a way to handle childrens children

for example

```csharp

public class Dimension
{
  public int height {get; set;}
  public int width {get;set;}
}

public class Square
{
  public Dimension Dimension {get;set;}
}
WithParametersFromObject(new Square(){Dimension = new Dimension {height = 20, width = 20}}) 
// should resolve as /?dimension.height=20&dimension.width=20
```

hint: this could be a recursive function ;)

Reply:
Yep I can probably get this done in a few days.